File: src/models/subject-averages.ts
import type { GradeValue } from "./grade-value";
import type { Subject } from "./subject";

export type SubjectAverages = Readonly<{
  /** students average in the subject */
  student?: GradeValue;
  /** classes average in the subject */
  class_average: GradeValue;
  /** highest average in the class */
  max: GradeValue;
  /** lowest average in the class */
  min: GradeValue;
  /** maximum amount of points */
  outOf?: GradeValue;
  /** the default maximum amount of points */
  defaultOutOf?: GradeValue;
  /** subject the average is from */
  subject: Subject;
  /** background color of the subject */
  backgroundColor: string;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/SubjectAverages.kt
package ink.literate.pawnote.models

data class SubjectAverages(
    /** students average in the subject */
    val student: GradeValue? = null,
    /** classes average in the subject */
    val classAverage: GradeValue,
    /** highest average in the class */
    val max: GradeValue,
    /** lowest average in the class */
    val min: GradeValue,
    /** maximum amount of points */
    val outOf: GradeValue? = null,
    /** the default maximum amount of points */
    val defaultOutOf: GradeValue? = null,
    /** subject the average is from */
    val subject: Subject,
    /** background color of the subject */
    val backgroundColor: String
)
